cloudinstall/utils.py

```diff
@@ -145,7 +145,37 @@ def sanitize_config_items(_cfg):
         return sanitize_config_items(cfg_cli_opts)
 
 
-def load_charms():
+def load_ext_charms(plug_path, charm_modules):
+    """ Load external charms from plugin directory
+
+    :param plug_path: Top level dir housing 'charms/'
+    :param charm_modules: Existing charms usually import by load_charms()
+    """
+    if os.path.exists(plug_path):
```

Comment:
I vote for at least a log message if not an exception (so the UI shows an error dialog), if the plug_path doesn't exist.

Then as a style point, we can avoid nesting lots of code and exit early:
```
if not os.path.exists(plug_path):
    raise Exception("Non-existent plugin path '{}' specified.".format(plug_path))

# rest of code is one less tab stop indented and we have more space for long lines:
try:
```